k9mail/src/main/java/com/fsck/k9/fragment/MessageListAdapter.java

```diff
@@ -163,6 +171,97 @@ public void bindView(View view, Context context, Cursor cursor) {
         String subject = MlfUtils.buildSubject(cursor.getString(SUBJECT_COLUMN),
                 fragment.getString(R.string.general_no_subject), threadCount);
 
+        TextView separatorText = (TextView) view.findViewById(R.id.separatorText);
+        TextView separatorDate = (TextView) view.findViewById(R.id.separatorDate);
+
+        // MessageInfoHolder message = (MessageInfoHolder) getItem(
+        // cursor.getPosition() );
+
+        boolean showSeparators = false;
+
+        if ((mSortType == Account.SortType.SORT_DATE) ||
+                (mSortType == Account.SortType.SORT_ARRIVAL)) {
+            showSeparators = true;
+        }
+
+        if (showSeparators) {
+            int pos = cursor.getPosition();
+            int[] categorizedMessages = fragment.getCategorizedMessages();
+            int dateCategory = categorizedMessages[pos];
+
+            if (mSortType == Account.SortType.SORT_DATE) {
+                displayDate = DateUtils.getRelativeTimeSpanString(context, cursor.getLong(DATE_COLUMN));
+            } else if (mSortType == Account.SortType.SORT_ARRIVAL) {
+                displayDate = DateUtils.getRelativeTimeSpanString(context, cursor.getLong(INTERNAL_DATE_COLUMN));
+            }
+
+            if (!datesDone) {
+                dateCategories[1] = mContext.getString(R.string.message_list_separator_today);
+                dateCategories[2] = mContext.getString(R.string.message_list_separator_yesterday);
+                dateCategories[3] = mContext.getString(R.string.message_list_separator_day_before_yesterday);
+
+                // dateCategories[3] = (String)
+                // android.text.format.DateFormat.format("EEEE",
+                // message.compareDate );
+
+                dateCategories[11] = mContext.getString(R.string.message_list_separator_last_week);
```

Comment:
Did you consider https://developer.android.com/reference/android/text/format/DateUtils.html